python/whylogs/core/datatypes.py

```diff
@@ -107,6 +107,26 @@ def _do_match(cls, dtype_or_type: Any, maybe_type: Optional[Any]) -> bool:
         return False
 
 
+class _PlaceHolder:
+    pass
+
+
+class Tensor(DataType[_PlaceHolder]):
```

Comment:
`DataType` doesn't like `np.ndarray` as an argument. Pandas sees tensors as `object`, so resolving by type only works for rows :(